cosmos/providers/dbt/render.py

```diff
@@ -196,79 +197,78 @@ def render_project(
-                    dbt_class="DbtTest",
-                ),
-                upstream_entity_ids=[run_task.id],
-                arguments=test_args,
-            )
-            entities[test_task.id] = test_task
-            # make the group
-            model_group = Group(
-                id=f"{model_name}",
-                entities=[run_task, test_task],
-            )
-            entities[model_group.id] = model_group
-            base_group.add_entity(entity=model_group)
-
-        # all other non-run tasks don't need to be grouped with test tasks
-        else:
-            entities[model_name] = run_task
-            base_group.add_entity(entity=run_task)
+        test_task = Task(
+            id=f"{model_name}_test",
+            operator_class=calculate_operator_class(
+                execution_mode=execution_mode,
+                dbt_class="DbtTest",
+            ),
+            upstream_entity_ids=[run_task.id],
+            arguments=test_args,
+        )
+        entities[test_task.id] = test_task
+        # make the group
+        model_group = Group(
+            id=f"{model_name}",
+            entities=[run_task, test_task],
+        )
+        entities[model_name] = model_group
+        base_group.add_entity(entity=model_group)
 
     # add dependencies now that we have all the entities
-    for model_name, model in itertools.chain(project.models.items(), project.snapshots.items(), project.seeds.items()):
+    for model_name, model in models_to_render.items():
+        # Only add dependencies if model in `entities`
         upstream_deps = model.config.upstream_models
         for upstream_model_name in upstream_deps:
-            try:
+            if upstream_model_name in entities:
                 dep_task = entities[upstream_model_name]
                 entities[model_name].add_upstream(dep_task)
-            except KeyError:
+            else:
                 logger.error(f"Dependency {upstream_model_name} not found for model {model}")
+
     if test_behavior == "after_all":
         # make a test task
         test_task = Task(
-            id=f"{dbt_project_name}_test",
+            id="dbt_test",
             operator_class=calculate_operator_class(
                 execution_mode=execution_mode,
                 dbt_class="DbtTest",
             ),
-            arguments={**task_args, **operator_args},
+            arguments={
+                **task_args,
+                **operator_args,
+                "profile_args": profile_args,
+                "profile_name": profile_name,
+                "target_name": target_name,
+                "on_warning_callback": on_warning_callback,
+            },
         )
         entities[test_task.id] = test_task
 
         # add it to the base group
         base_group.add_entity(test_task)
 
         # add it as an upstream to all the models that don't have downstream tasks
-        # since we don't have downstream info readily available, we have to iterate
```

Comment:
This change fix the bug: no dependence to test_task when we set select or exclude.